content/dvacaty_paty_sraz.md

```diff
@@ -0,0 +1,58 @@
+Title: Dvacátý pátý sraz - po delší době
+Date: 2019-05-14 18:00:00
+Modified: 2019-05-14 18:00:00
+Author: Anežka Müller
+
+Dnešní sraz proběhl po delší pauze, kdy Petr nebyl v Brně, ale v kódu jsme se hodně posunuly.
+Stkání jsme tedy začly shrnutím, co hra nyní umí. Součástí backendu je teď důležitá a komplexní funkce `apply_all_effects`, která zařídí, že proběhne herní kolo. Tedy střídavě se v daném pořadí provedou všechny naplánované akce z registru robota (efekty karet) a efekty políček. 
+Při optimalizaci jejího kódu jsme narazily na dvě věci. První z nich je vyvolání vyjímky. Je třeba si vždy dobře rozmyslet, jakou vyjímku vyvoláváme a najakém místě. Specifikace by měla být co nejpřesnější, aby se nám nestalo, že vyjímka bude odchytávat jiné chyby nebo chyby vzniklé na jiném místě, než to, které chceme ošetřit. 
+Druhá připomínka se nám v kódu stále vrací, a to je pojmenovávání funkcí. Je to dobrá praxe, kterou bychom si měly vštípit a držet se jí, protože výrazně zlepšuje přehlednost kódu. Pokud pojmenováváme nějakou funkci, jméno by mělo odpovídat tomu, co daná funkce dělá. Například máme-li funkci `sort_robots_by_cards_priority`, lze očekávat, že vezme nějaký seznam robotů a seřadí jej dle daného parametru. V našem případě ale funkce vytvářela nový seznam a teprve ten řadila, jméno tedy není zcela příhodné.
+Karolina také výrazně pokročila s testovacím frameworkem, který nyní umí načíst příkazy ze všech adresářů, zpracovávat různé podmínky a po provedení příkazů zkontrolovat novou polohu a směr robota a porovnat je s konečnými políčky. 
+Katka také rozpracovala jedno z vylepšení, které se s námi táhne již delší dobu, a to je načítání obrázků ve frontendu. Poměrně velká výpočetní náročnost hry byla dána tím, že jsme obrázky ve frontendu načítaly pořád dokola, přestože to není potřeba, stačilo by je načíst jen jednou. Nejobtížnějsí částí se nakonec ukázalo rozhodnout, jaké zvolíme názvy políček, jejichž obrázky se načítají, aby byly dostatečně výstižné, takže i zde jsme se dostaly zpět k pravidlu správného pojmenovávání. 
```

Comment:
Jsou tam dvě věty, každá začíná stejně: `Karolina také`, `Katka také` - je to záměr? Pokud ne, hlasuju pro přepsání jedné formulace.

Reply:
Upraveno.